tidy3d/components/simulation.py

```diff
@@ -1058,6 +1061,42 @@ def _validate_monitor_size(self) -> None:
                 f"a maximum of {MAX_SIMULATION_DATA_SIZE_GB:.2f}GB are allowed."
             )
 
+    def _validate_modes_size(self) -> None:
+        """Warn if mode sources or monitors have a large number of points."""
+
+        def warn_mode_size(monitor: AbstractModeMonitor, msg_header: str, custom_loc: List):
+            """Warn if a mode component has a large number of points."""
+            num_cells = np.prod(self.discretize_monitor(monitor).num_cells)
+            if num_cells > WARN_MODE_NUM_CELLS:
+                consolidated_logger.warning(
+                    msg_header + f"has a large number ({num_cells:1.2e}) of grid points. "
+                    "This can lead to solver slow-down and increased cost. "
+                    "Consider making the size of the component smaller, as long as the modes "
+                    "of interest decay by the plane boundaries."
+                )
+
+        with log as consolidated_logger:
+            for src_ind, source in enumerate(self.sources):
+                if isinstance(source, ModeSource):
+                    # Make a monitor so we can call ``discretize_monitor``
+                    monitor = FieldMonitor(
+                        center=source.center,
+                        size=source.size,
+                        name="tmp",
+                        freqs=[source.source_time.freq0],
+                        colocate=False,
+                    )
+                    msg_header = f"Mode source at sources[{src_ind}] "
+                    custom_loc = ["sources", src_ind]
+                    warn_mode_size(monitor=monitor, msg_header=msg_header, custom_loc=custom_loc)
+
+        with log as consolidated_logger:
+            for monitor in self.monitors:
+                if isinstance(monitor, AbstractModeMonitor):
+                    msg_header = f"Mode monitor '{monitor.name}' "
+                    custom_loc = ["monitors", monitor.name]
```

Comment:
let's make it `custom_loc = ["monitors", mnt_ind]` so that it's consistent with pydantic style

Reply:
Ok